src/test/java/net/sf/jabref/gui/keyboard/KeyBindingsDialogViewModelTest.java

```diff
@@ -0,0 +1,175 @@
+        KeyEvent shortcutKeyEvent = prepareNewKeyBind(KeyBinding.ABBREVIATE, "J", KeyCode.J, true, true, true, false);
+        assertFalse(keyBindingsPreferences.checkKeyCombinationEquality(KeyBinding.ABBREVIATE, shortcutKeyEvent));
+        model.grabKey(shortcutKeyEvent);
+
+        KeyEvent shortcutKeyEvent1 = prepareNewKeyBind(KeyBinding.UNMARK_ENTRIES, "X", KeyCode.X, true, false, false,
+                false);
+        assertFalse(keyBindingsPreferences.checkKeyCombinationEquality(KeyBinding.UNMARK_ENTRIES, shortcutKeyEvent1));
+        model.grabKey(shortcutKeyEvent1);
+
+        model.saveKeyBindings();
+
+        assertTrue(keyBindingsPreferences.checkKeyCombinationEquality(KeyBinding.ABBREVIATE, shortcutKeyEvent));
+        assertTrue(keyBindingsPreferences.checkKeyCombinationEquality(KeyBinding.UNMARK_ENTRIES, shortcutKeyEvent1));
+    }
+
+    @Test
+    public void testSetAllKeyBindingsToDefault() {
+        KeyBindingViewModel viewModel = new KeyBindingViewModel(KeyBinding.ABBREVIATE,
+                KeyBinding.ABBREVIATE.getDefaultBinding());
+        model.getSelectedKeyBinding().set(new TreeItem<>(viewModel));
+        KeyEvent shortcutKeyEvent = new KeyEvent(KeyEvent.KEY_PRESSED, "C", "C", KeyCode.C, true, true, true, false);
+
+        assertFalse(keyBindingsPreferences.checkKeyCombinationEquality(KeyBinding.ABBREVIATE, shortcutKeyEvent));
+
+        model.grabKey(shortcutKeyEvent);
+        model.saveKeyBindings();
+
+        assertTrue(keyBindingsPreferences.checkKeyCombinationEquality(KeyBinding.ABBREVIATE, shortcutKeyEvent));
+
+        model.resetKeyBindingsToDefault();
+        model.saveKeyBindings();
+
+        assertFalse(keyBindingsPreferences.checkKeyCombinationEquality(KeyBinding.ABBREVIATE, shortcutKeyEvent));
+    }
+
+    @Test
+    public void testSingleKeyBindingToDefault() {
+        KeyBindingViewModel viewModel = new KeyBindingViewModel(KeyBinding.ABBREVIATE,
+                KeyBinding.ABBREVIATE.getDefaultBinding());
+        model.getSelectedKeyBinding().set(new TreeItem<>(viewModel));
+        KeyEvent shortcutKeyEvent = new KeyEvent(KeyEvent.KEY_PRESSED, "C", "C", KeyCode.C, true, true, true, false);
+
+        assertFalse(keyBindingsPreferences.checkKeyCombinationEquality(KeyBinding.ABBREVIATE, shortcutKeyEvent));
+
+        model.grabKey(shortcutKeyEvent);
+        model.saveKeyBindings();
+
+        assertTrue(keyBindingsPreferences.checkKeyCombinationEquality(KeyBinding.ABBREVIATE, shortcutKeyEvent));
+
+        viewModel.resetToDefault(keyBindingRepository);
+        model.saveKeyBindings();
+
+        assertFalse(keyBindingsPreferences.checkKeyCombinationEquality(KeyBinding.ABBREVIATE, shortcutKeyEvent));
+    }
+
+    private KeyEvent prepareNewKeyBind(KeyBinding binding, String text, KeyCode code, boolean shiftDown,
+            boolean controlDown, boolean altDown, boolean metaDown) {
+        KeyBindingViewModel bindViewModel = new KeyBindingViewModel(binding, binding.getDefaultBinding());
```

Comment:
I would split this method, so only create a shortcut for setting the selected keybinding.